include/pybind11/detail/common.h

```diff
@@ -154,6 +154,8 @@
 // C4505: 'PySlice_GetIndicesEx': unreferenced local function has been removed (PyPy only)
 #  pragma warning(disable: 4505)
 #  if defined(_DEBUG) && !defined(Py_DEBUG)
+// fix for vs2022: https://github.com/pybind/pybind11/issues/3477
+#    include <crtdefs.h>
```

Comment:
This needs to be enabled only for msvc 22+. It might add a new dependency on an optional component. Plus it will help reduce the impact if something breaks.

Reply:
This doesn't add any dependencies which are not already there. Pybind11 is already dependent on standard C library.